src/traces/bar/defaults.js

```diff
@@ -61,22 +62,49 @@ function supplyDefaults(traceIn, traceOut, defaultColor, layout) {
 function crossTraceDefaults(fullData, fullLayout) {
     var traceIn, traceOut;
 
-    function coerce(attr) {
-        return Lib.coerce(traceOut._input, traceOut, attributes, attr);
+    function coerce(attr, dflt) {
+        return Lib.coerce(traceOut._input, traceOut, attributes, attr, dflt);
     }
 
-    if(fullLayout.barmode === 'group') {
-        for(var i = 0; i < fullData.length; i++) {
-            traceOut = fullData[i];
+    for(var i = 0; i < fullData.length; i++) {
+        traceOut = fullData[i];
+
+        if(traceOut.type === 'bar') {
+            traceIn = traceOut._input;
+            // `marker.cornerradius` needs to be coerced here rather than in handleStyleDefaults()
+            // because it needs to happen after `layout.barcornerradius` has been coerced
+            var r = coerce('marker.cornerradius', fullLayout.barcornerradius);
+            traceOut.marker.cornerradius = validateCornerradius(r);
 
-            if(traceOut.type === 'bar') {
-                traceIn = traceOut._input;
+            if(fullLayout.barmode === 'group') {
                 handleGroupingDefaults(traceIn, traceOut, fullLayout, coerce);
             }
         }
     }
 }
 
+// Returns a value equivalent to the given cornerradius value, if valid;
+// otherwise returns`undefined`.
+// Valid cornerradius values must be either:
+//   - a numeric value (string or number) >= 0, or
+//   - a string consisting of a number >= 0 followed by a % sign
+// If the given cornerradius value is a numeric string, it will be converted
+// to a number.
+function validateCornerradius(r) {
+    var validR = false;
+    if(isNumeric(r)) {
+        r = +r;
+        if(r >= 0) validR = true;
+    } else if(typeof r === 'string' && r.slice(-1) === '%') {
+        r = +r.slice(0, -1);
+        if(r >= 0) {
+            r += '%';
+            validR = true;
```

Comment:
```suggestion
        if(isNumeric(r.slice(0, -1)) {
            return r;
```
Best to consistently use `isNumeric`, or we end up with odd situations, like `"%"` would be valid since `+"" === 0` 😱 - doesn't seem like we want that.

I guess I could also ask about trailing whitespace, which we DO accept for numeric strings. I don't feel too strongly about maintaining that here, but for maximum consistency we could choose to accept `"\t 50 % \n"` as a valid percentage.

Reply:
Fixed! I added a `r = r.trim()` step as well, is that an appropriate way of handling trailing whitespace?

Reply:
> Fixed! I added a `r = r.trim()` step as well, is that an appropriate way of handling trailing whitespace?

Yes. Resolved.